src/Workspaces/Remote/Core/RemoteHostOptions.cs

```diff
@@ -48,6 +53,21 @@ public static bool IsServiceHubProcessServerGC(HostWorkspaceServices services)
                 || services.GetService<IExperimentationService>()?.IsExperimentEnabled(WellKnownExperimentNames.OOPServerGC) == true;
         }
 
+        /// <summary>
+        /// Determines whether ServiceHub out-of-process execution is enabled for Roslyn.
+        /// </summary>
+        public static bool IsUsingServiceHubOutOfProcess(HostWorkspaceServices services)
+        {
+            var optionService = services.GetRequiredService<IOptionService>();
+            if (Environment.Is64BitOperatingSystem && optionService.GetOption(OOP64Bit))
```

Comment:
Not that it will matter in dev17, but we might want to use OOP if OS is 32bit. Also should remove 64bit from the comment below

Reply:
I'll simplify this condition.

Reply:
I'll send the simplification in a follow-up to avoid delaying this PR.